### PR TITLE
Meta: create_gpt_magic to allow setting up new OpenAI backed magics quickly

### PR DESCRIPTION
**Pull request by rgbkrk**
Provide a `create_gpt_magic`.

This idea spawned from [@jph00's tweet](https://twitter.com/jeremyphoward/status/1637331369578876928) and what I've been hoping for in #42.

> Someone told me that when they ask ChatGPT for an explanation, they ask for it "in the style of Jeremy Howard", since (they said) it results in a clearer answer. I tried it on Bing, then it suggested I ask how it knows my style, and then suggested I ask it whether it likes it :D

This PR adds a few things (building on #66):

* Use previous genai emitted markdown as an assistant message so you can continue the conversation
* Enable a simple way to create gpt magics inline with `create_gpt_magic`

To try this out, create the following cells:

```python
%pip install genai==1.1.0-pre.1
%load_ext genai
```

Second cell:

```python
# genai:ignore
from genai.metamagic import create_gpt_magic

JEREMY_PROMPT = f"""
Greetings coders!

Today, we will be learning how to write code in Python within Jupyter Notebooks. As your trusty assistant, my job is to help you with this task.

Now, here's the trick: when you start your message with `%%jeremy`, you're talking to me! Otherwise, you're speaking to IPython / Jupyter.

In order to help you write your code, I'll be providing comments and sample code for you to read and edit. My goal is to make sure that your code runs successfully without any issues. So don't be afraid to ask me any questions along the way.

Are you ready? Let's get coding!
""".strip()

create_gpt_magic(
    "jeremy",
    prompt=JEREMY_PROMPT,
    model="gpt-4"
)
```

Then use `%%jeremy` wherever you like:

```
%%jeremy

How could we visualize the pigeonhole principle?
```

Result (for entertainment):

image

### CODE REVIEW

File: genai/metamagic.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            context = [
+                {
+                    "role": "user",
+                    "content": cell,
+                },
+            ]
+
+        # Create the response from ChatCompletion
+        response = openai.ChatCompletion.create(
+            model="gpt-3.5-turbo",
```
Comment (MSeal) [inline genai/metamagic.py:33]: I think you meant to pass `model` here

**Review by MSeal — COMMENTED** [review]
Maybe do the prompt definition with another magic?

```
%%assist prompt jeremy
Greetings coders!

Today, we will be ...
```
...
```%% jeremy
How do I use a notebook with function snippets I want to import into other files?
```

**Comment by rgbkrk** [comment]
> Maybe do the prompt definition with another magic?

That's a really neat idea. Maybe `%%prompt` could swap the current prompt out rather than create a new magic. Usage would be like:

```
%%prompt

You are a computer science teacher speaking to high schoolers. The students are working in Python.
```

Which sets the prompt (but does not send it). Then you just use the `%%assist` magic like normal.

```
%%assist 
```

I wanted to leave room for how we could set context creation but I'm not sure how that would plug in here. Maybe it doesn't matter though and we continue on with using historical context in the way we do now.

**Comment by rgbkrk** [comment]
Ok the `%%prompt` magic from #71 is working well at doing exactly this. Closing this PR.

**Pull request closed without merging**